### PR TITLE
Add date range field support

### PR DESCRIPTION
**Pull request by edetec**

### CODE REVIEW

File: src/ng-xform/date-range-field/date-range-field.component.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    multi: true
+  }],
+})
+export class DateRangeFieldComponent extends BaseDynamicFieldComponent<DateRangeField> implements AfterViewInit, AfterContentInit,
+  ControlValueAccessor {
+  config: Partial<BsDatepickerConfig>;
+  componentControl = new FormControl();
+  private input: HTMLInputElement;
+
+  _onChange = (value: any) => { };
```
Comment (igortg) [inline src/ng-xform/date-range-field/date-range-field.component.ts:32]: Why you need that?

File: src/ng-xform/date-range-field/date-range-field.component.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    super();
+    this.componentControl.valueChanges.subscribe((val: any) => {
+      // replay changes from view to the form value
+      this._onChange(val);
+    });
+  }
+
+  ngAfterContentInit() {
+    this.componentControl.setValidators(this.field.validators);
+    this.config = Object.assign({}, {
```
Comment (igortg) [inline src/ng-xform/date-range-field/date-range-field.component.ts]: Why `Object.assign`?

File: src/ng-xform/date-range-field/date-range-field.component.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  writeValue(obj: any): void {
+    if (obj instanceof Array) {
+      this.componentControl.setValue(obj.map(item => new Date(item)));
+    } else {
+      this.componentControl.setValue(obj);
+    }
+  }
+  registerOnChange(fn: any): void {
```
Comment (igortg) [inline src/ng-xform/date-range-field/date-range-field.component.ts]: Aren't we using prettier here?

**Review by lvoliveira — APPROVED** [review]

File: src/ng-xform/date-range-field/date-range-field.component.ts (hunk 3 above)
Comment (edetec) [reply src/ng-xform/date-range-field/date-range-field.component.ts]: No, just tslinter

File: src/ng-xform/date-range-field/date-range-field.component.ts (hunk 2 above)
Comment (edetec) [reply src/ng-xform/date-range-field/date-range-field.component.ts]: I'll remove it

File: src/ng-xform/date-range-field/date-range-field.component.ts (hunk 1 above)
Comment (edetec) [reply src/ng-xform/date-range-field/date-range-field.component.ts:32]: To avoid error if it is called before registerOnChange

**Review by igortg — APPROVED** [review]

**Review by lincrosenbach — APPROVED** [review]

**Pull request merged**